financial-connections/src/main/java/com/stripe/android/financialconnections/navigation/Destination.kt

```diff
@@ -0,0 +1,200 @@
+        route = Pane.MANUAL_ENTRY.value,
+        composable = { ManualEntryScreen() }
+    )
+
+    object AttachLinkedPaymentAccount : NoArgumentsDestination(
+        route = Pane.ATTACH_LINKED_PAYMENT_ACCOUNT.value,
+        composable = { AttachPaymentScreen() }
+    )
+
+    object NetworkingLinkSignup : NoArgumentsDestination(
+        route = Pane.NETWORKING_LINK_SIGNUP_PANE.value,
+        composable = { NetworkingLinkSignupScreen() }
+    )
+
+    object NetworkingLinkLoginWarmup : NoArgumentsDestination(
+        route = Pane.NETWORKING_LINK_LOGIN_WARMUP.value,
+        composable = { NetworkingLinkLoginWarmupScreen() }
+    )
+
+    object NetworkingLinkVerification : NoArgumentsDestination(
+        route = Pane.NETWORKING_LINK_VERIFICATION.value,
+        composable = { NetworkingLinkVerificationScreen() }
+    )
+
+    object NetworkingSaveToLinkVerification : NoArgumentsDestination(
+        route = Pane.NETWORKING_SAVE_TO_LINK_VERIFICATION.value,
+        composable = { NetworkingSaveToLinkVerificationScreen() }
+    )
+
+    object LinkAccountPicker : NoArgumentsDestination(
+        route = Pane.LINK_ACCOUNT_PICKER.value,
+        composable = {
+            LinkAccountPickerScreen()
+        },
+    )
+
+    object LinkStepUpVerification : NoArgumentsDestination(
+        route = Pane.LINK_STEP_UP_VERIFICATION.value,
+        composable = { LinkStepUpVerificationScreen() }
+    )
+
+    object Reset : NoArgumentsDestination(
+        route = Pane.RESET.value,
+        composable = { ResetScreen() }
+    )
+
+    object ManualEntrySuccess : Destination(
+        route = Pane.MANUAL_ENTRY_SUCCESS.value,
+        params = listOf(KEY_MICRODEPOSITS, KEY_LAST4),
+        screenBuilder = { ManualEntrySuccessScreen(it) }
+    ) {
+
+        fun argMap(
+            microdepositVerificationMethod: MicrodepositVerificationMethod,
+            last4: String?
+        ): Map<String, String?> = mapOf(
+            KEY_MICRODEPOSITS to microdepositVerificationMethod.value,
+            KEY_LAST4 to last4
+        )
+
+        fun microdeposits(backStackEntry: NavBackStackEntry): MicrodepositVerificationMethod =
+            backStackEntry.arguments
+                ?.getString(KEY_MICRODEPOSITS)
+                ?.let { value ->
+                    MicrodepositVerificationMethod.values().firstOrNull { it.value == value }
+                } ?: MicrodepositVerificationMethod.UNKNOWN
+
+        fun last4(backStackEntry: NavBackStackEntry): String? =
+            backStackEntry.arguments?.getString(KEY_LAST4)
+
+        override fun invoke(args: Map<String, String?>): String = route.appendParams(
+            KEY_MICRODEPOSITS to args[KEY_MICRODEPOSITS],
+            KEY_LAST4 to args[KEY_LAST4]
+        )
+    }
+
+    companion object {
+        const val KEY_MICRODEPOSITS = "microdeposits"
+        const val KEY_LAST4 = "last4"
+    }
+}
+
+internal fun String.appendParams(vararg params: Pair<String, Any?>): String {
```

Comment:
could we rename this to `appendParamValues` ?

Reply:
https://github.com/stripe/stripe-android/pull/7250/commits/bf2b2e8186bb2be2fcd69f4fbe84f3e39d936055